### PR TITLE
[MSAPI-56] Created solution architecture to cover  Get Signing Link function

### PR DESCRIPTION
**Pull request by IvanBil**

### CODE REVIEW

File: SignNow.Net/Interface/IOAuth2Service.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SignNow.Net.Interface
+{
+    public interface IOAuth2Service
+    {
+        Task<Token> GetTokenAsync(string login, string password, Scope scope, CancellationToken cancellationToken = default(CancellationToken));
+        Task<Token> GetTokenAsync(string code, Scope scope, CancellationToken cancellationToken = default(CancellationToken));//TODO: https://pdffiller.atlassian.net/wiki/spaces/API2/pages/911769793/Grant+type+authorization+code
```
Comment (AlexNDRmac) [inline SignNow.Net/Interface/IOAuth2Service.cs:11]: Pls remove comment. You can configure VS to exclude such things in commit

File: SignNow.Net/_Internal/Constants/ApiUrl.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+using System;
+
+namespace SignNow.Net._Internal.Constants
+{
+    static class ApiUrl
+    {
+        public static Uri ApiBaseUrl = new Uri("https://api.signnow.com");
```
Comment (AlexNDRmac) [inline SignNow.Net/_Internal/Constants/ApiUrl.cs:7]: for a future - it will be better if we'll use env/config files for this constants

**Review by AlexNDRmac — COMMENTED** [review]
My suggestion: it will be better, if we will add explicit doc block for every public method, especially for Interfaces. Each params must contains details helping to exclude any dual meanings or things understanding, even if it obvious.

File: SignNow.Net/Interface/IOAuth2Service.cs (hunk 1 above)
Comment (IvanBil) [reply SignNow.Net/Interface/IOAuth2Service.cs:11]: @AlexNDRmac why? This is hint for developer where to find information on implementation of the method.

**Comment by IvanBil** [comment]
> My suggestion: it will be better, if we will add explicit doc block for every public method, especially for Interfaces. Each params must contains details helping to exclude any dual meanings or things understanding, even if it obvious.

@AlexNDRmac  I agree but currently we don't actually implement anything. This is just a high level design that could change and I suggest comments to be written by developer of the method

**Review by ZolotarovLeonid — APPROVED** [review]
Wonderfull!

**Review by AlexNDRmac — APPROVED** [review]

**Review by horobets — APPROVED** [review]

**Review by KhodanAndrii — APPROVED** [review]

**Pull request merged**